lib/spack/spack/cmd/testinstall.py

```diff
@@ -0,0 +1,181 @@
+#
+# This file is part of Spack.
+# Written by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://scalability-llnl.github.io/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License (as published by
+# the Free Software Foundation) version 2.1 dated February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from external import argparse
+import xml.etree.ElementTree as ET
+import itertools
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import *
+
+import spack
+import spack.cmd
+
+description = "Build and install packages"
+
+def setup_parser(subparser):
+    #subparser.add_argument(
+    #    '-i', '--ignore-dependencies', action='store_true', dest='ignore_deps',
+    #    help="Do not try to install dependencies of requested packages.")
+    
+    subparser.add_argument(
+        '-j', '--jobs', action='store', type=int,
+        help="Explicitly set number of make jobs.  Default is #cpus.")
+        
+    #always false for test
+    #subparser.add_argument(
+    #    '--keep-prefix', action='store_true', dest='keep_prefix',
+    #    help="Don't remove the install prefix if installation fails.")
+    
+    #always true for test
+    #subparser.add_argument(
+    #    '--keep-stage', action='store_true', dest='keep_stage',
+    #    help="Don't remove the build stage if installation succeeds.")
+    
+    subparser.add_argument(
+        '-n', '--no-checksum', action='store_true', dest='no_checksum',
+        help="Do not check packages against checksum")
+    subparser.add_argument(
+        '-v', '--verbose', action='store_true', dest='verbose',
+        help="Display verbose build output while installing.")
+    
+    #subparser.add_argument(
+    #    '--fake', action='store_true', dest='fake',
+    #    help="Fake install.  Just remove the prefix and touch a fake file in it.")
+    
+    subparser.add_argument(
+        'output', help="test output goes in this file")
+    
+    subparser.add_argument(
+        'packages', nargs=argparse.REMAINDER, help="specs of packages to install")
+
+
+class JunitResultFormat(object):
+    def __init__(self):
+        self.root = ET.Element('testsuite')
+        self.tests = []
+        
+    def addTest(self, buildId, passed=True, buildInfo=None):
```

Comment:
Can you use `add_test` and `write_to` format for methods, like the rest of Spack?